corehq/apps/accounting/models.py

```diff
@@ -1601,13 +1605,19 @@ def renew_subscription(self, note=None, web_user=None,
                 "There was an issue renewing your subscription. Someone "
                 "from Dimagi will get back to you shortly."
             )
+
+        if new_version.plan.is_annual_plan:
+            new_date_end = self.date_end.replace(year=self.date_end.year + 1)
+        else:
+            new_date_end = None
```

Comment:
This block is not itself feature flagged, but the only way for an annual plan version to get passed in is with the feature flag enabled.

Reply:
Just thinking out loud, and I don't think this is really necessary but curious to hear your thoughts on it. Does it seem valuable to add an assertion that an annual plan is passed in only if that feature flag is enabled? That would effectively do what your comment in this PR is doing but for future code readers as well. Not sure if it is worth it though.

Reply:
I don't think you're wrong, and that's something I should consider if doing something similar with feature flags going forward. At the same time, assuming all looks as good on production as it does on staging, this will only be feature flagged for the next week.